.github/workflows/slack.yml

```diff
@@ -24,33 +24,42 @@ jobs:
       - uses: actions/checkout@v2
         with:
           ref: main
-      
+
       # Install dotenv python package to load tokens in script
       - name: Install dependencies
         run: |
           python3 -m pip install --upgrade pip
-          pip3 install python-dotenv
-            
+          # pip3 install python-dotenv
+
       # Runs a script using the runners shell
       - name: Run a user extraction script
-        working-directory: ./metrics/slack          
+        working-directory: ./metrics/slack
         env:
-          SLACK_KEY: ${{ secrets.SLACK_KEY }}
-          
+          MINIO: ${{ secrets.MINIO }}
```

Comment:
@venkuppu-chn  Just suggestion not sure if it's feasible. As I understand these are tokens/cookies for fetching data from Slack endpoints. Can we generate them runtime from user-name. We can take user-name as input and no need to store any secrets. 

@johnbent  Hope it's ok to use secrets with these names. Just confirming as we cleared Product-specific words from repositories as a part of contentious word removal activity.

Reply:
@shailesh-vaidya It is feasible. But, username & password authentication is deprecated due to security reasons not only in slack, so I suggest opting API based authentication.  Moreover, for the script to run, one has to join all the workspace to get metrices count. So, I suggest opting service account with their token to get information without having to join. Also, workflow is not designed for individuals.

Reply:
@venkuppu-chn  Thanks for the info. So can we use a single API token for all workspaces to reduce secrets count to one.  I agree we should use a service account for fetching info.

Reply:
@shailesh-vaidya, I have verified it is not possible, we should require the dedicated token for each workspace.